### PR TITLE
remove CYGNUS_SERVICE_PORT env var

### PR DESCRIPTION
**Pull request by AlvaroVega**
https://github.com/telefonicaid/fiware-cygnus/issues/1761#issuecomment-559090072

### CODE REVIEW

File: docker/cygnus-ngsi/Dockerfile [hunk 1]
```diff
@@ -341,6 +340,4 @@ COPY cartodb_keys.conf ${FLUME_HOME}/conf/
 # Define the entry point
 ENTRYPOINT ["/cygnus-entrypoint.sh"]
 
-# Ports used by cygnus-ngsi
-EXPOSE ${CYGNUS_SERVICE_PORT} ${CYGNUS_API_PORT}
```
Comment (fgalan) [inline docker/cygnus-ngsi/Dockerfile]: Looking to this removal, should  `ENV CYGNUS_API_PORT "5080"` line be also removed?

Comment (AlvaroVega) [reply docker/cygnus-ngsi/Dockerfile]: That exposes in Dockerfile does not define all ports that tipically we use. Docker-compose.yml has all ports, and we can remove `expose`

Comment (jason-fox) [reply docker/cygnus-ngsi/Dockerfile]: Personally I think this change is the wrong way up, the EXPOSE should list all potential ports you can expose

```Dockerfile
# Ports used by cygnus-ngsi
EXPOSE ${CYGNUS_SERVICE_PORT} ${CYGNUS_API_PORT}  ${CYGNUS_MYSQL_SERVICE_PORT} \
  ${CYGNUS_MONGO_SERVICE_PORT} ${CYGNUS_CKAN_SERVICE_PORT} ${CYGNUS_HDFS_SERVICE_PORT} \
  ${CYGNUS_CARTO_SERVICE_PORT} ${CYGNUS_POSTGRESQL_SERVICE_PORT} \
  ${CYGNUS_ORION_SERVICE_PORT} ${CYGNUS_POSTGIS_SERVICE_PORT} \
  ${CYGNUS_ELASTICSEARCH_SERVICE_PORT} ${CYGNUS_ARCGIS_SERVICE_PORT}
```

`EXPOSE` in the Dockerfile is primarily a **documentation** mechanism - it is stating what you could expose using `-P` for example. Which ports you *actually* open and assign is up to you.

see: https://www.ctl.io/developers/blog/post/docker-networking-rules/ as an example

Comment (jason-fox) [reply docker/cygnus-ngsi/Dockerfile]: By the way, is `CYGNUS_ORION_SERVICE_PORT` supposed to be the same as `CYGNUS_MYSQL_SERVICE_PORT` or should it be 5056?

**Comment by jason-fox** [comment]
Depending on how you decide to proceed with this, I'd like to suggest that the  [install_with_docker.md](https://github.com/telefonicaid/fiware-cygnus/blob/master/doc/cygnus-ngsi/installation_and_administration_guide/install_with_docker.md) is updated to align with the preferred best practice.

either:

1. Keep `CYGNUS_SERVICE_PORT` as an ENV, and a reference in the docker documentation that the `CYGNUS_SERVICE_PORT` need to be exposed and match the relevant  `CYGNUS_XXXXX_SERVICE_PORT`.  Also add that the `CYGNUS_XXXX_HOST` must be set to generate the conf. Also give an example in the code.

2. Remove the `CYGNUS_SERVICE_PORT` as an ENV, and update the `expose` command to list all potential sinks as shown in https://github.com/telefonicaid/fiware-cygnus/pull/1764#discussion_r351470448, Add a reference and example  in the docker documentation that the `CYGNUS_XXXXX_SERVICE_PORT` need to be exposed and `CYGNUS_XXXX_HOST` must be set

Either of these would fix https://github.com/telefonicaid/fiware-cygnus/issues/1761 as the docker ports issue is merely down to a lack of documentation, and the actual documentation which needs to be written will depend on the evolution of this PR.

**Pull request closed without merging**